web/packages/teleport/src/services/userPreferences/userPreferences.ts

```diff
@@ -108,3 +131,36 @@ export function makeDefaultUserClusterPreferences(): ClusterUserPreferences {
     },
   };
 }
+
+export function convertUserPreferences(
+  preferences: UserPreferences
+): BackendUserPreferences {
+  return {
+    ...preferences,
+    clusterPreferences: {
+      pinnedResources:
+        preferences.clusterPreferences?.pinnedResources?.resourceIds ?? [],
+    },
```

Comment:
I'd spread `clusterPreferences` too:
```
    clusterPreferences: {
      ...preferences.clusterPreferences,
      pinnedResources:
        preferences.clusterPreferences?.pinnedResources?.resourceIds ?? [],
    }
```
If someone adds a new property, we won't skip it. The same for `convertBackendUserPreferences`.